src/rpm/compressor.rs

```diff
@@ -107,37 +135,70 @@ impl Compressor {
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
 pub enum CompressionWithLevel {
     None,
+    #[cfg(feature = "zstd-compression")]
     Zstd(i32),
+    #[cfg(feature = "gzip-compression")]
     Gzip(u32),
+    #[cfg(feature = "xz-compression")]
     Xz(u32),
+    #[cfg(feature = "bzip2-compression")]
     Bzip2(u32),
 }
 
 impl CompressionWithLevel {
+    // This function is used to check if a compression type is zstd in builder.rs:760. If zstd is
+    // not enabled, this function will be unused and therefor dead code.
+    #[cfg_attr(not(feature = "zstd-compression"), allow(dead_code))]
     pub(crate) fn compression_type(&self) -> CompressionType {
         match self {
             Self::None => CompressionType::None,
+            #[cfg(feature = "gzip-compression")]
             Self::Gzip(_) => CompressionType::Gzip,
+            #[cfg(feature = "zstd-compression")]
             Self::Zstd(_) => CompressionType::Zstd,
+            #[cfg(feature = "xz-compression")]
             Self::Xz(_) => CompressionType::Xz,
+            #[cfg(feature = "bzip2-compression")]
             Self::Bzip2(_) => CompressionType::Bzip2,
         }
     }
 }
 
 impl Default for CompressionWithLevel {
+    // The preference of the default compression type is listed in decending order and dependent on
```

Comment:
I actually submitted a PR just now to default to Zstd.  Now that EL7 is end of life, we can do that without causing too many issues.

Reply:
You may feel free to adopt that change and the changelog entry in this PR, for the sake of avoiding conflicts later.